pkg/apis/fleet.cattle.io/v1alpha1/imagescan_types.go

```diff
@@ -62,6 +62,13 @@ type ImageScanSpec struct {
 	// +nullable
 	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
 
+	// CABundleRef can be given the name of a ConfigMap containing a TLS CA to
+	// be used when connecting to the Image Registry.
+	//
+	// The ConfigMap should have a key `ca.crt` with a PEM encoded CA bundle.
+	// +nullable
+	CABundleRef *corev1.LocalObjectReference `json:"caBundleRef,omitempty"`
```

Comment:
We could use https://github.com/rancher/fleet/blob/main/pkg/apis/fleet.cattle.io/v1alpha1/bundledeployment_types.go#L285 since it's already there.

The GitRepo uses a []byte for the PEM:
https://github.com/rancher/fleet/blob/main/pkg/apis/fleet.cattle.io/v1alpha1/gitrepo_types.go#L89-L91

A separate config map seems better, as it's easier to create. Using the same approach as the Gitrepo would be consistent? Not sure if that's important here. Ideally they would use the same
config map?

Reply:
I followed the SecretRef above which also uses the corev1.LocalObjectReference ?

Reply:
Yeah, as I said in chat, all repositories within the same registry would presumably use the same CA, duplicating this across all `ImageScan` resources feels wasteful, never mind that it adds maintenance cost.

I think we could add support to `GitRepo` to allow you to reference the same `ConfigMap`.